data_structures/src/transaction.rs

```diff
@@ -161,9 +170,19 @@ impl VTTransaction {
         VTTransaction { body, signatures }
     }
 
-    /// Returns the byte size that a transaction will have on the wire
+    /// Returns the size that a value transfer transaction will have on the wire
+    #[allow(clippy::cast_possible_truncation)]
     pub fn size(&self) -> u32 {
-        u32::try_from(self.to_pb().write_to_bytes().unwrap().len()).unwrap()
+        // VT_weight = N*INPUT_SIZE + M*OUTPUT_SIZE*gamma
+        let inputs_len = self.body.inputs.len() as u32;
```

Comment:
Clippy warns you, if someone manages to send a transaction with 2^32 inputs (not possible right now, but maybe using compression?), `inputs_len` will be truncated to 0.

```suggestion
        let inputs_len = u32::try_from(self.body.inputs.len()).unwrap_or(u32::MAX);
```

Reply:
Or you can use an early return, whatever looks better:

```suggestion
        let inputs_len = self.body.inputs.len();
        let inputs_len = match u32::try_from(inputs_len) { Ok(x) => x, Err(e) => return u32::MAX, }
```